### DISCUSSION TITLE
Re-installing zfsbootmenu

### DISCUSSION THREAD

**Discussion by Sithuk** [zbm-dev/zfsbootmenu#524] [General]
I am having trouble updating zfsbootmenu on a legacy ubuntu install. I use a custom config so a drop in pre-compiled efi bundle is not a solution.

I have re-installed zfsbootmenu on an ubuntu server install. I compiled from the 2.2.2 zfsbootmenu source code on github. "Generate-zbm --showver" returns 2.2.2 to confirm the update worked. I have run "update-initramfs -c -k all" to update the initramfs and then "generate-zbm --debug" to update zfsbootmenu. Unfortunately the zfsbootmenu accessible during boot does not appear to have  updated. How would I confirm the zfsbootmenu version from within zfsbootmenu? 

You can see the code I am using to update zfsbootmenu in the ubuntu zfsbootmenu install script "reinstall-zbm" function. Link below.
https://github.com/Sithuk/ubuntu-server-zfsbootmenu

**Comment by zdykstra** [comment]
There isn't a reasonable way to see the version from within ZFSBootMenu right now. Your best option, though, is to drop to the recovery shell and run `zreport` to compare the kernel and ZFS versions. That'll change between old and new images.

**Reply by Sithuk** [reply]
Should the zbm UI and help files get refreshed with installing a new version? Should "generate-zbm" create an entirely new version of Zbm for rEFInd to load?

**Reply by classabbyamp** [reply]
generate-zbm will entirely replace the existing image

**Comment by zdykstra** [comment]
You'll need to double check that your stage 1 bootloader (refind, efiboot entry, etc) points to the correct file. We can't debug your local system, so this is something that you'll have to do on your own.

**Reply by Sithuk** [reply]
Thank you zdykstra, I think rEFInd was pre-selecting the boot files in an old non-maintained folder in /boot/efi/EFI. I removed the folder and rEFInd  appears to have auto-selected the correct files to boot from. So not an issue with zfsbootmenu.

Now I just need to figure out why I seem to be missing the screen summary GUI at the top. The one that has "[ Boot Environments ] - Snapshots - Kernels - Pool Status" with the square brackets switching as the GUI is navigated.

**Comment by zdykstra** [comment]
The version of `fzf` that Ubuntu is providing is too old. You need at a minimum [0.35.0](https://github.com/zbm-dev/zfsbootmenu/blob/master/zfsbootmenu/install-helpers.sh#L91-L96)
